.github/workflows/test.yml

```diff
@@ -7,25 +7,35 @@ on:
     branches: [ main, master, develop ]
 
 jobs:
-  test:
-    runs-on: ubuntu-latest
+
+  cdm_jupyterhub_tests:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - python-version: "3.11"
 
     steps:
-    - uses: actions/checkout@v4
-    
-    - name: Set up Python 3.11
+    - name: Repo checkout
+      uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5
       with:
-        python-version: '3.11'
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
         
     - name: Install dependencies
+      shell: bash
       run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
+        export UV_PROJECT_ENVIRONMENT="${pythonLocation}"
+        uv sync --locked
         
     - name: Run tests
-      run: |
-        PYTHONPATH=src pytest -v --cov=src --cov-report=xml tests
+      shell: bash
+      run: PYTHONPATH=src pytest --cov=src --cov-report=xml tests
```

Comment:
```suggestion
      run: PYTHONPATH=src uv run pytest --cov=src --cov-report=xml tests
```

Reply:
no sure. Maybe it's installed globally. But it works.